src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -1481,22 +1483,23 @@ namespace winrt::TerminalApp::implementation
                                                                     const PasteFromClipboardEventArgs eventArgs)
     {
         co_await winrt::resume_foreground(Dispatcher(), CoreDispatcherPriority::High);
-
-        TerminalPage::PasteFromClipboard(eventArgs);
+        
+        TerminalPage::PasteFromClipboard(eventArgs, _settings->GlobalSettings().RemoveTabsOnPaste());
     }
 
     // Function Description:
     // - Copies and processes the text data from the Windows Clipboard.
     //   Does some of this in a background thread, as to not hang/crash the UI thread.
     // Arguments:
     // - eventArgs: the PasteFromClipboard event sent from the TermControl
-    fire_and_forget TerminalPage::PasteFromClipboard(PasteFromClipboardEventArgs eventArgs)
+    fire_and_forget TerminalPage::PasteFromClipboard(PasteFromClipboardEventArgs eventArgs, bool removeTabs)
```

Comment:
I'd probably _not_ add this as a parameter to this function, and instead to the `weakThis` dance (mentioned above) in this method to just look up the value after the `resume_background` call.

Reply:
We tried this, but we weren't able to get it work for the same reason that we added it as a parameter in the first place--PasteFromClipboard is a static function, so get_weak() doesn't work in it. Is there some other way to get around it?